charts/sourcegraph/values.yaml

```diff
@@ -401,7 +401,9 @@ gitserver:
     allowPrivilegeEscalation: false
     runAsUser: 100
     runAsGroup: 101
-    readOnlyRootFilesystem: true
+    # -- Read-only filesystem not supported for the gitserver container,
+    # see [sourcegraph/issues/34012](https://github.com/sourcegraph/sourcegraph/issues/34012) for more information
+    readOnlyRootFilesystem: false
```

Comment:
For gitserver - maybe instead of disabling readOnly, we can try adding an emptydir `/tmp` volume mount? Similar to [searcher here](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/ace2ed4f57202b90d4b01c84e4f2d0ffba930587/charts/sourcegraph/templates/searcher/searcher.Deployment.yaml#L92-L93).

Reply:
Tested this out, looks good on my end.